Modules/System/Math/src/MathImpl.Codeunit.al

```diff
@@ -0,0 +1,63 @@
+codeunit 711 "Math Impl."
+{
+    Access = Internal;
+
+    procedure BitwiseAnd(IntegerValue1: Integer; IntegerValue2: Integer) Result: Integer
+    var
+        BitPosition: Integer;
+        BitIndex: Integer;
+    begin
+        InitBitwiseValues(BitPosition, Result);
+
+        for BitIndex := to 32 do begin
+            if (IntegerValue1 mod 2 = 1) and (IntegerValue2 mod 2 = 1) then
+                UpdateBitwiseResult(BitPosition, Result);
```

Comment:
For readability, it'd be preferred to change Result to a local variable and exit(Result) at the end.

Reply:
Once the tests and this change are in, I'll process this today.

Reply:
One more thing, I think these functions will only work for positive numbers. If that's the case, either add validation that the parameters cannot be negative or ensure these will work for negative numbers too.

Reply:
I will add that to my test and add validation as needed.